src/test/java/io/mathdojo/QuestionFunctionTest.java

```diff
@@ -1,39 +1,130 @@
 
+import com.microsoft.azure.functions.ExecutionContext;
+
 public class QuestionFunctionTest {
 
-	@Test
-	public void test() {
-		Question question = mock(Question.class);
-		when(question.getQuestionTitle()).thenReturn("test");
-		QuestionFunction qf = new QuestionFunction();
-		qf.repository = mock(MathDojoQuestionRepository.class);
+	private Question question;
+	private QuestionFunction qf = new QuestionFunction();
+	private MathDojoQuestionRepository repo = mock(MathDojoQuestionRepository.class);
+	private MathDojoTopicRepository trepo = mock(MathDojoTopicRepository.class);
+	private Topic topic = new Topic();
+	Map<String, String> headers;
+	Map<String, String> queryParams;
+
+	@Before
+	public void setUp() {
+		question = new Question();
+		question.setQuestionTitle("test");
+		question.setDifficulty("easy");
+		question.setId("test");
+		qf.repository = repo;
+		qf.tRepository = trepo;
 		List<Question> questionList = new ArrayList<Question>();
 		questionList.add(question);
-		when(qf.repository.findAll()).thenReturn(questionList);
-		Question result = qf.getQuestion().apply(new GetObject());
-		assertEquals(result.getQuestionTitle(), "test");
+		when(repo.findByQuestionTitle("test")).thenReturn(questionList);
+		when(repo.findById("test")).thenReturn(Optional.of(question));
+		when(repo.findAllById(new ArrayList<String>(Arrays.asList("test")))).thenReturn(questionList);
+		topic.setId("test");
+		topic.setName("test");
+		topic.setTitle("test");
+		topic.setQuestions(new ArrayList<String>(Arrays.asList("test")));
+		List<Topic> topicList = new ArrayList<Topic>();
+		topicList.add(topic);
+		when(trepo.findByTopicTitle("test")).thenReturn(topicList);
+		when(trepo.findById("test")).thenReturn(Optional.of(topic));
+		headers = new HashMap<>();
+		queryParams = new HashMap<>();
+
 	}
 
 	@Test
+	public void testQuestionAndTopicFunctions() {
+		Question getQuestionResult = qf.getQuestion().apply(question);
+		assertEquals(getQuestionResult, question);
+		qf.updateQuestion().accept(question);
+		verify(repo, times(1)).save(question);
+		Question getQuestionByIdResult = qf.getQuestionById().apply(question);
+		assertEquals(getQuestionByIdResult, question);
+		qf.deleteQuestion().accept(question);
+		verify(repo, times(1)).deleteById("test");
+		Topic getTopicResult = qf.getTopic().apply(topic);
+		assertEquals(getTopicResult, topic);
+		qf.updateTopic().accept(topic);
+		verify(trepo, times(1)).save(topic);
+		Topic getTopicByIdResult = qf.getTopicById().apply(topic);
+		assertEquals(getTopicByIdResult, topic);
+		qf.deleteTopic().accept(topic);
+		verify(trepo, times(1)).deleteById("test");
+		assertEquals(qf.getQuestions().apply(topic).get(0), question);
+
+	}
+
 	@Ignore
-	public void start() throws Exception {
-		AzureSpringBootRequestHandler<GetObject, Question> handler = new AzureSpringBootRequestHandler<>(
+	@Test
+	public void testQuestionHandler() throws URISyntaxException {
```

Comment:
I've read through this and I'm not sure what we're trying to test. Is it that the handleRequest ends up callling the getQuestion function?

Reply:
yeah that's the intent, it feels out of scope though so I'm thinking of remving it and expanding the other tests